lib/testscript_engine/testscript_runnable.rb

```diff
@@ -170,6 +175,20 @@ def load_fixtures
     end
   end
 
+  def load_profiles
+    script.profile.each do |profile|
+      next warning(:no_static_profile_id) unless profile.id
+      next warning(:no_static_profile_reference) unless profile.reference
+
+      profile_server = FHIR::Client.new("")
+      response = profile_server.send(:get, profile.reference, { 'Content-Type' => 'json' })
+      next if response.response[:code] == "200"
+
+      profiles[profile.id] = FHIR.from_contents(response.response[:body].to_s)
```

Comment:
> 

I kinda follow this idea, then load_fixtures may be better to be moved outside runnable as well? I thought load_fixture and load_profile are similar in the sense of when and how loaded and used..

Reply:
I am not sure I follow this. Two issues:
1. What is multiple TestScripts have duplicate but inconsistent definitions of profiles? For example:
TestScrpit A
```
  "profile": [
    {
      "id": "patient-profile",
      "reference": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"
    }
  ]
```

TestScrpit B
```
  "profile": [
    {
      "id": "patient-profile",
      "reference": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient_different"
    }
  ]
```

Since they have same profile ID, when the engine stores them into profiles Hash, the latter will overwrite the former?

2. Codewise, where do we put load_profiles()?

engine.load_input
<< here?
engine.make_runnables

Reply:
Moving profile loading outside the runnable to the engine level sounds like a potentially nice optimization, but Jaehoon's duplicate name for different profiles example suggests that it'll take some work and will be better left for a future task. I would suggest creating a low priority issue for this optimization for investigation later.